test/benchmark_utils/test_benchmark_utils.py

```diff
@@ -0,0 +1,829 @@
+            <torch.utils.benchmark.utils.common.Measurement object at 0xXXXXXXXXXXXX>
+            x + 1: scalar_add
+              Median: 10.02 us
+              IQR:    0.67 us (9.67 to 10.34)
+              100 measurements, 10000 runs per measurement, 1 thread"""
+        )
+
+        assert_reprs_match(
+            MockTimer(
+                stmt="""
+                    with torch.no_grad():
+                        y = x + 1
+                """,
+                setup="setup_fn()",
+                sub_label="scalar_add"
+            ).blocked_autorange(min_run_time=10),
+            """
+            <torch.utils.benchmark.utils.common.Measurement object at 0xXXXXXXXXXXXX>
+            stmt: (scalar_add)
+              with torch.no_grad():
+                  y = x + 1
+
+            setup: setup_fn()
+              Median: 10.02 us
+              IQR:    0.67 us (9.67 to 10.34)
+              100 measurements, 10000 runs per measurement, 1 thread"""
+        )
+
+        assert_reprs_match(
+            MockTimer(
+                stmt="""
+                    with torch.no_grad():
+                        y = x + 1
+                """,
+                setup="""
+                    x = torch.ones((1,), requires_grad=True)
+                    for _ in range(5):
+                        x = x + 1.0""",
+                sub_label="scalar_add",
+                description="Multi-threaded scalar math!",
+                num_threads=16,
+            ).blocked_autorange(min_run_time=10),
+            """
+            <torch.utils.benchmark.utils.common.Measurement object at 0xXXXXXXXXXXXX>
+            stmt: (scalar_add)
+              with torch.no_grad():
+                  y = x + 1
+
+            Multi-threaded scalar math!
+            setup:
+              x = torch.ones((1,), requires_grad=True)
+              for _ in range(5):
+                  x = x + 1.0
+
+              Median: 10.02 us
+              IQR:    0.67 us (9.67 to 10.34)
+              100 measurements, 10000 runs per measurement, 16 threads"""
+        )
+
+    @slowTest
+    @unittest.skipIf(IS_WINDOWS, "Valgrind is not supported on Windows.")
+    def test_collect_callgrind(self):
+        @torch.jit.script
+        def add_one(x):
+            return x + 1
+
+        timer = benchmark_utils.Timer(
+            "y = add_one(x) + k",
+            setup="x = torch.ones((1,))",
+            globals={"add_one": add_one, "k": 5}
+        )
+
+        # Don't collect baseline to speed up unit test by ~30 seconds.
+        stats = timer.collect_callgrind(number=1000, collect_baseline=False)
+        counts = stats.counts(denoise=False)
+
+        self.assertIsInstance(counts, int)
+        self.assertGreater(counts, 0)
+
+    def test_manipulate_callgrind_stats(self):
+        stats_no_data, stats_with_data = load_test_example()
+
+        self.assertEqual(stats_no_data.counts(), 8869966)
+        self.assertEqual(stats_no_data.counts(denoise=True), 8728096)
```

Comment:
Should have instructions for how to update the test numbers (and check that they're right!) if you refresh the fixture